### PR TITLE
Add request specs for authentication flows

### PR DESCRIPTION
**Pull request by santiagorodriguez96**
## Motivation

Right now we don't have specific tests for our strategies apart from our system tests, which because of its nature don't really cover all the edge cases. This is leading to weird behaviors such as the one described in #88.

The goal of this PR is to add request specs to test the different authentication flows, which the different edge case, in order to try to test all the strategies and how they work together.

This could in term allow us to stop relying too much in our system tests which have been behaving a little bit flaky.

## Details

This PR adds two request specs: `spec/requests/devise/passkey_authentication_spec.rb` (for testing authentication with passkeys) and `spec/requests/devise/two_factor_authentication_spec.rb` (for testing authentication with email/password and 2FA).

### CODE REVIEW

File: spec/requests/devise/passkey_authentication_spec.rb [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+  let(:password) { "password123" }
+  let(:user) { Account.create!(email: "test@example.com", password: password) }
+  let(:origin) { WebAuthn.configuration.allowed_origins.first }
+  let(:client) { WebAuthn::FakeClient.new(origin) }
+
+  def create_passkey_for(account, fake_client)
+    challenge = WebAuthn.configuration.encoder.encode(SecureRandom.random_bytes(32))
+    raw_credential = fake_client.create(challenge: challenge)
+    webauthn_credential = WebAuthn::Credential.from_create(raw_credential)
+    webauthn_credential.verify(challenge)
```
Comment (brauliomartinezlm) [inline spec/requests/devise/passkey_authentication_spec.rb]: Does it really make sense to verify the credential if it's just for test setup?

Comment (santiagorodriguez96) [reply spec/requests/devise/passkey_authentication_spec.rb]: Yeah my bad. It's not needed at all 👍
